site_scons/site_tools/go_builder.py

```diff
@@ -49,6 +48,17 @@ def _scan_go_file(node, env, _path):
     return includes
 
 
+def get_min_go_version():
+    """Get go minimum version from go.mod"""
+    go_mod_path = os.path.join(Dir('#').abspath, "src", "control", "go.mod")
+    with open(go_mod_path, 'r') as f:
+        for line in f:
+            if line.startswith('go '):  # e.g. "go 1.21"
+                parts = line.split()
+                return get_go_version("go" + parts[1])
+    raise Exception("no minimum go version found in go.mod")
```

Comment:
@jolivier23 might remember quicker than me but there's a `scons` exception we cal call here I think.  Either way the error needs to be resolved before this can be landed.

Reply:
Reading this through again, there's a scons configure check going on here but the configure check itself is also only called from this file and if the configure check fails then the build stops with `Exit(1)` (a scons function).

What I think you need to do here is return `None` here, and then when calling this function later on check for `None` and call `context.Result` with the error string and return 0.